dask/dataframe/io/parquet/arrow.py

```diff
@@ -265,149 +91,540 @@ def _index_in_schema(index, schema):
         return False  # No index to check
 
 
-class ArrowEngine(Engine):
-    @classmethod
-    def read_metadata(
-        cls,
-        fs,
-        paths,
-        categories=None,
-        index=None,
-        gather_statistics=None,
-        filters=None,
-        split_row_groups=True,
-        **kwargs,
-    ):
-        # Define the dataset object to use for metadata,
-        # Also, initialize `parts`.  If `parts` is populated here,
-        # then each part will correspond to a file.  Otherwise, each part will
-        # correspond to a row group (populated below)
-        parts, dataset = _determine_dataset_parts(
-            fs, paths, gather_statistics, filters, kwargs.get("dataset", {})
+def _get_dataset_object(paths, fs, filters, dataset_kwargs):
+    """ Generate a ParquetDataset object
+    """
+    if len(paths) > 1:
+        # This is a list of files
+        base, fns = _analyze_paths(paths, fs)
+        proxy_metadata = None
+        if "_metadata" in fns:
+            # We have a _metadata file. PyArrow cannot handle
+            #  "_metadata" when `paths` is a list. So, we shuld
+            # open "_metadata" separately.
+            paths.remove(fs.sep.join([base, "_metadata"]))
+            fns.remove("_metadata")
+            proxy_metadata = (
+                pq.ParquetDataset(
+                    fs.sep.join([base, "_metadata"]),
+                    filesystem=fs,
+                    filters=filters,
+                    **dataset_kwargs,
+                )
+                .pieces[0]
```

Comment:
To be sure: this is the metadata file itself, contains the metadata pieces of all the files, but to get that metadata we must refer to one piece?

Reply:
Yes,  I believe this is the way the `ParquetDataset` works - It breaks the dataset into "pieces", and for the _metadata file, there is only one "piece". We can also do:

```python
with fs.open(fs.sep.join([base, "_metadata"]), mode="rb") as fil:
    proxy_metadata = pq.ParquetFile(fil).metadata
```

but I believe the missing `filters` argument means we cannot handle filtering on partitioned columns.  That said, I am not sure there is much of usecase for passing in a list of files for a partitioned dataset anyway.  If @jorisvandenbossche is availbale, he may be able to advise.

Reply:
> I am not sure there is much of usecase for passing in a list of files for a partitioned dataset

I have parquet files with a directory partitioning. Each directory contains multiple historical/duplicate versions of the same data (it's partitioned by the `value_date` rather than the `issue_date`). It's certainly possible to read in every single file and then filter to get the latest data but for performance we record which file is the latest so rather than reading 1000's of redundant files and then post-filtering we can read in only the data we're interested in. So, for this usecase it would be good to allow specifying the files to read.

Reply:
> but I believe the missing filters argument means we cannot handle filtering on partitioned columns.

But the `_metadata` file shouldn't be filtered out anyway, I think? So I think your code snippet above might be a bit cleaner (although doesn't matter much in practice)